pkg/distributor/distributor.go

```diff
@@ -774,16 +781,32 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 
 	totalSamples := validatedFloatSamples + validatedHistogramSamples
 	totalN := totalSamples + validatedExemplars + len(validatedMetadata)
-	if !d.ingestionRateLimiter.AllowN(now, userID, totalN) {
+
+	nhRateLimited := false
+	if limits.NativeHistogramsIngestionRate != math.MaxFloat64 {
```

Comment:
We can remove the `if`. Lets leave the library do its job. I dont expect it to be resource intensive

Reply:
Thanks updated. 
After removing this if condition, the Push tests failed for global strategy (when histograms were enabled). That was b/c the global rate strategy divide inf by num of distributors, and then its not a special value any more for rate lib.